components/builder-originsrv/src/migrations/origin_channels.rs

```diff
@@ -124,173 +124,191 @@ pub fn migrate(migrator: &mut Migrator) -> SrvResult<()> {
                         RETURN;
                     END
                     $$ LANGUAGE plpgsql STABLE"#,
     )?;
     migrator.migrate(
-        "originsrv-v2",
-        r#"CREATE OR REPLACE FUNCTION get_origin_channel_package_latest_v2 (
+        "originsrv",
+        r#"CREATE OR REPLACE FUNCTION get_origin_channel_package_v3 (
                     op_origin text,
                     op_channel text,
                     op_ident text,
-                    op_target text
+                    op_account_id bigint
                  ) RETURNS SETOF origin_packages AS $$
                     BEGIN
                         RETURN QUERY SELECT op.*
                           FROM origin_packages op
                           INNER JOIN origin_channel_packages ocp on ocp.package_id = op.id
                           INNER JOIN origin_channels oc on ocp.channel_id = oc.id
                           INNER JOIN origins o on oc.origin_id = o.id
-                          WHERE o.name = op_origin
+                          WHERE op.ident = op_ident
+                          AND o.name = op_origin
                           AND oc.name = op_channel
-                          AND op.ident LIKE (op_ident  || '%')
-                          AND op.target = op_target;
+                          AND (op.visibility='public' OR (op.visibility IN ('private', 'hidden') AND op.origin_id IN (SELECT origin_id FROM origin_members WHERE account_id = op_account_id)));
                         RETURN;
                     END
                     $$ LANGUAGE plpgsql STABLE"#,
     )?;
     migrator.migrate("originsrv",
-                     r#"CREATE OR REPLACE FUNCTION get_origin_package_channels_for_package_v1 (
-                    op_ident text
+                     r#"CREATE OR REPLACE FUNCTION get_origin_package_channels_for_package_v3 (
+                    op_ident text,
+                    op_account_id bigint
                  ) RETURNS SETOF origin_channels AS $$
                     BEGIN
                         RETURN QUERY SELECT oc.*
                             FROM origin_channels oc INNER JOIN origin_channel_packages ocp ON oc.id = ocp.channel_id
                             INNER JOIN origin_packages op ON op.id = ocp.package_id
-                            WHERE op.ident=op_ident
+                            WHERE op.ident = op_ident
+                            AND (op.visibility='public' OR (op.visibility IN ('private', 'hidden') AND op.origin_id IN (SELECT origin_id FROM origin_members WHERE account_id = op_account_id)))
                             ORDER BY oc.name;
                         RETURN;
                     END
                     $$ LANGUAGE plpgsql STABLE"#)?;
     migrator.migrate(
         "originsrv",
-        r#"CREATE OR REPLACE FUNCTION get_origin_channel_package_latest_v3 (
+        r#"DROP FUNCTION IF EXISTS get_origin_channel_package_v1(text, text, text)"#,
+    )?;
+    migrator.migrate(
```

Comment:
We could probably leave out the drops for the old functions - I think the thought was we'd clean up the code with removal of the old functions, not the DB...

Reply:
@reset said he thought dropping the old functions was a good idea.  We definitely need a better way to manage this kind of cruft long term though.

Reply:
We can drop the functions as the service lives as long as no active server is using this function any longer. Basically Live is our end of line in the CI process and if it doesn’t use the function you can drop it :)